torch/csrc/utils/future.h

```diff
@@ -61,25 +61,23 @@ class TORCH_API Future final {
     return std::move(value_);
   }
 
-  void markCompleted(T value) {
+  // Marks the future complete only if it hasn't been marked completed already.
+  void markCompletedIfNeeded(T value) {
     std::unique_lock<std::mutex> lock(mutex_);
-    TORCH_CHECK(!completed_);
-    // Set value first as completed_ is accessed without lock
-    value_ = std::move(value);
-    completed_ = true;
-
-    // Move callbacks to a vector on the stack so we can access it without
-    // holding a lock
-    std::vector<std::function<void(void)>> cbs;
-    cbs.swap(callbacks_);
-    lock.unlock();
-    finished_cv_.notify_all();
-    // There is no need to protect callbacks_ with the lock.
-    for (auto& callback : cbs) {
-      callback();
+    if (completed_) {
+      LOG(INFO) << "Skipping marking the Future completed since it has " <<
```

Comment:
Do we really want to log?
I realize that `setErrorIfNeeded()` logs, though we might log if the caller wants it by returning true/false whether we were able to markCompleted.
 
(Or, if we do log, can we at least not call operator<<() twice?)